src/appleseed/renderer/kernel/rendering/generic/genericsamplerenderer.cpp

```diff
@@ -253,6 +254,22 @@ namespace
                 if (!shading_point_ptr->hit_surface())
                     break;
 
+                const Material* material = shading_point_ptr->get_material();
```

Comment:
Add a short comment summarizing what this code block does.

My main question here is: is this code where it ought to be (that is, between checking whether we hit the environment and whether we've reached full opacity)?

Reply:
I thought it's first thing to check - if it's matte surface or not, that's why I put it right after surface hit check.

Reply:
I think I should have done compound the same way as `CompositeEmissionClosure::process_closure_tree`. I'll fix it tomorrow.